### ISSUE TITLE
StringAssert.Contains fails when comparing strings with single ticks, using "uz" culture 

### ISSUE THREAD

**Issue by grrblt** [nunit/nunit#3401]
I ran into this issue when testing my project for all cultures that were installed on my machine. I'm not 100% sure that it's your fault (could be the fault of CultureInfo, or it could be a feature of the Uzbek language). But here is an example that breaks in a clean project:

```
using System.Globalization;
using System.Threading;
using NUnit.Framework;

namespace ClassLibrary1
{
    [TestFixture]
    public class Class1
    {
        [Test]
        public void PrintCulture()
        {
            CultureInfo savedCurrent = Thread.CurrentThread.CurrentCulture;
            CultureInfo savedDefault = CultureInfo.DefaultThreadCurrentCulture;
            CultureInfo culture = new CultureInfo("uz"); //uzbekistan

            Thread.CurrentThread.CurrentCulture = culture;
            CultureInfo.DefaultThreadCurrentCulture = culture;

            string withTicks = "a string containing the word 'blarg' with ticks";
            string withoutTicks = "a string containing the word blarg without ticks";
            string blarg = "blarg";
            
            //works
            Assert.That(withTicks.Contains(blarg));
            Assert.That(withoutTicks.Contains(blarg));
            StringAssert.Contains(blarg, withoutTicks);

            //breaks
            StringAssert.Contains(blarg, withTicks);
            
            Thread.CurrentThread.CurrentCulture = savedCurrent;
            CultureInfo.DefaultThreadCurrentCulture = savedDefault;
        }
    }
}
```

I'm using Rider 2019.2.2 and have NUnit 3.12.0 and NUnit3TestAdapter 3.15.1 installed.

**Comment by mikkelbu** [comment]
The test passes on my machine, but I ran the test using VS Test Explorer.

**Comment by CharliePoole** [comment]
@grrblt What's the precise failure message?

**Comment by grrblt** [comment]
> @grrblt What's the precise failure message?

```
ClassLibrary1.Class1.PrintCulture

  Expected: String containing "blarg"
  But was:  "a string containing the word 'blarg' with ticks"

   at ClassLibrary1.Class1.PrintCulture() in /home/edvin/RiderProjects/Solution2/ClassLibrary1/Class1.cs:line 30
```

I've now debugged my way to `bool Matches(string actual)` in SubstringConstraint.cs. It uses StringComparison.CurrentCulture as I'd expect, and everything looks reasonable until it calls `return actual.IndexOf(expected, actualComparison) >= 0;` But for some reason the IndexOf returns -1. So I guess this is probably not a problem with NUnit as it first seemed.

**Comment by grrblt** [comment]
Okay, I have tracked down the source of the problem, which turns out to be - a feature of the Uzbek language.

I found out on wikipedia that the Uzbek alphabet has a letter which, when written with latin script, is **gʻ** (g with a "modifier letter turned comma"). My string includes **g'** (g with apostrophe) and apparently my system can't tell the difference between these two symbols, so it interprets the **g'* as one distinct letter that's different from a regular **g**.

All the tests work if I swap "blarg" for something that doesn't end with a **g**, so it's definitely not NUnit's fault.
